pymodbus/client/mixin.py

```diff
@@ -551,14 +551,21 @@ def convert_to_registers(
         """Convert int/float/str to registers (16/32/64 bit).
 
         :param value: value to be converted:
-        :param data_type: data type to convert to
+        :param data_type: data type to be encoded as registers
         :returns: List of registers, can be used directly in e.g. write_registers()
+        :raises TypeError: when there is a mismatch between data_type and value
         """
-        if data_type == cls.DATATYPE.STRING:
-            byte_list = value.encode()  # type: ignore[union-attr]
+        if "STRING" in data_type.name:
```

Comment:
I think it is a bad idea to have an extra string "STRING", why do you see that as a better option ?

Reply:
I did it to be similar to `if "INT" in data_type.name:` below.

Reply:
How about dropping the "INT" check, and change "STRING" to cls.DATATYPE.STRING.name, then I am ready to merge.

Reply:
Hmm, OK but I am on mobile and can't do this today

Reply:
I can do it, if you are ok with that.

Reply:
I am converting unittest to pytest, which is 10 tons of silly editing...so something to break the rhythm is welcome.